tests/models/test_transformers.py

```diff
@@ -153,3 +153,34 @@ def test_phi3_chat_fixed(phi3_model: models.Model):
         lm += gen(name="five", max_tokens=10)
 
     assert "5" in lm["five"]
+
+
+def test_phi3_newline_chat():
+    from guidance.models import Transformers
+    lm = Transformers("microsoft/Phi-3-mini-4k-instruct",
+        trust_remote_code=True,
+    )
+    lm += "You are a counting bot. Just keep counting numbers."
+    with user():
+        lm += "1\n2\n3\n4\n"
+    with assistant():
+        lm += "\n" + gen(name="five", max_tokens=1)
+        lm += "\n" + gen(name="six", max_tokens=1)
+    
+    assert True
```

Comment:
I assume that without this fix, something above will throw an exception?

Reply:
Yes the newlines cause special issues with phi-3's tokenizer.

Reply:
Can you add a comment to that effect?